new-integration-tests/src/test/java/oracle/weblogic/kubernetes/ItVoyagerSample.java

```diff
@@ -0,0 +1,298 @@
+    logger.info("Get a unique namespace for WebLogic domain");
+    assertNotNull(namespaces.get(2), "Namespace list is null");
+    domainNamespace = namespaces.get(2);
+
+    final String cloudProvider = "baremetal";
+    final boolean enableValidatingWebhook = false;
+
+    // install and verify Voyager
+    voyagerHelmParams =
+      installAndVerifyVoyager(voyagerNamespace, cloudProvider, enableValidatingWebhook);
+
+    // install and verify operator
+    installAndVerifyOperator(opNamespace, domainNamespace);
+
+    // create and verify one cluster domain
+    logger.info("Create domain and verify that it's running");
+    createAndVerifyDomain();
+  }
+
+  @AfterAll
+  void tearDown() {
+    // uninstall Voyager
+    if (voyagerHelmParams != null) {
+      assertThat(uninstallVoyager(voyagerHelmParams))
+          .as("Test uninstallVoyager returns true")
+          .withFailMessage("uninstallVoyager() did not return true")
+          .isTrue();
+    }
+  }
+
+  /**
+   * The test invokes a webapp to verify the Voyager is installed successfully and ready to use.
+   */
+  @Test
+  @DisplayName("Create the Voyager ingress and invoke the webapp to verify Voyager load balancer works")
+  @Slow
+  @MustNotRunInParallel
+  public void testVoyagerWorks() {
+    String ingressName = domainUid + "-ingress-host-routing";
+    String ingressServiceName = VOYAGER_CHART_NAME + "-" + ingressName;
+    String channelName = "tcp-80";
+
+    try {
+      // create Voyager ingress resource
+      Map<String, Integer> clusterNameMsPortMap = new HashMap<>();
+      clusterNameMsPortMap.put(clusterName, managedServerPort);
+      List<String>  hostNames =
+          installVoyagerIngressAndVerify(domainUid, domainNamespace, ingressName, clusterNameMsPortMap);
+
+      // wait 60 seconds for ingress service to be ready
+      Thread.sleep(60 * 1000);
+      logger.info("Getting node port for ingress service: " + ingressServiceName);
+
+      // get ingress service Nodeport
+      int ingressServiceNodePort = assertDoesNotThrow(()
+          -> getServiceNodePort(domainNamespace, ingressServiceName, channelName),
+              "Getting admin server node port failed");
+      logger.info("Node port for {0} is: {1} :", ingressServiceName, ingressServiceNodePort);
+
+      // invoke webapp
+      String curlRequest = String.format("curl --silent --show-error --noproxy '*' "
+          + "-H 'host: %s' http://%s:%s/sample-war/index.jsp",
+              hostNames.get(0), K8S_NODEPORT_HOST, ingressServiceNodePort);
+      logger.info("Exec curl command :" + curlRequest);
+      ExecResult result = ExecCommand.exec(curlRequest, true);
+      assertNotNull(result, "curl command returns null");
+      logger.info("curl command returns: \n{0}, \n{1}", result.stdout(), result.stderr());
+      assertTrue(result.stdout().contains("Hello World"), "Failed to invoke the webapp");
+    } catch (Exception ex) {
+      ex.printStackTrace();
```

Comment:
test should log the exception and fail on exception

Reply:
fixed